sync/syncer.go

```diff
@@ -255,9 +256,12 @@ func (s *Syncer) GetFullBlocks(blockIds []types.BlockID) []*types.Block {
 }
 
 func (s *Syncer) BlockSyntacticValidation(block *types.Block) ([]*types.AddressableSignedTransaction, []*types.ActivationTx, error) {
-	if err := s.confirmBlockValidity(block); err != nil {
-		s.Error("failed derefrencing block data %v %v", block.ID(), err)
-		return nil, nil, errors.New(fmt.Sprintf("failed derefrencing block data %v %v", block.ID(), err))
+	//todo remove this hack once identity genesis setup is implemented
+	if block.Layer().GetEpoch(s.LayersPerEpoch).IsGenesis() == false {
```

Comment:
There are two things that you can't validate on Genesis - whether or not the miner is active and the block's signature. However, you can still validate the block's eligibility.
So, you should check if the epoch is 0 (`IsGenesis` isn't good here since it also considers epoch 1 as Genesis) and pass this info to `BlockEligible` nd there skip the signature and activeness validation